stl/inc/filesystem

```diff
@@ -1686,23 +1686,33 @@ namespace filesystem {
         return false;
     }
 
-    _NODISCARD inline path path::lexically_relative(const path& _Base) const {
+    _NODISCARD inline path path::lexically_relative(const path& _Base_) const {
+        // N4868 [fs.path.gen]
+
         constexpr wstring_view _Dot     = L"."sv;
         constexpr wstring_view _Dot_dot = L".."sv;
 
+        // LWG-3699: `lexically_relative` on UNC drive paths (`\\?\C:\...`) results in a default-constructed value
+        // This avoids doing any unnecessary copies;
+        // the return value of `relative_path()` is lifetime extended if necessary.
+        const path& _This = _Is_drive_prefix_with_slash_slash_question(native()) ? relative_path() : *this;
+        const path& _Base =
+            _Is_drive_prefix_with_slash_slash_question(_Base_.native()) ? _Base_.relative_path() : _Base_;
+
         path _Result;
 
-        if (root_name() != _Base.root_name() || is_absolute() != _Base.is_absolute()
-            || (!has_root_directory() && _Base.has_root_directory()) || _Relative_path_contains_root_name(*this)
-            || _Relative_path_contains_root_name(_Base)) {
+        if (_This.root_name() != _Base.root_name() || _This.is_absolute() != _Base.is_absolute()
+            || (!_This.has_root_directory() && _Base.has_root_directory())
+            || (_Relative_path_contains_root_name(_This) || _Relative_path_contains_root_name(_Base))) {

```

Comment:
Question: Why are extra parens being introduced here? This is continuing a logical-OR chain so there are no precedence issues. The similarity in the conditions `_Relative_path_contains_root_name(_This)` and `_Relative_path_contains_root_name(_Base)` is clear, so there appears to be no need to group them with extra parens.

Reply:
I grouped them to follow the logic of the prose better, imo. The prose is written as:

If one of the following is true:
* A
* B
* C and D
* E or F

and so I prefer to write it as a list of or-separated boolean expressions, with each element of the list being fully contained, rather than transforming the final expression by splitting up `E or F` into separate elements. I can revert it, but I do believe it's more readable (one element being on one line, rather than split so that `(C and D) or E` are on one line with `F` being on another), and this doesn't change behavior at all.

Reply:
Sounds good, I'm fine with this change!